app_components: Center-align the shortcut keys in help menu.

whilstsomebody opened the pull request:
- Shortcut keys in help menu were not centered in light theme. This PR adds `text-align: center` for the keys in the light theme.
- We had different border-radius defined for light and dark theme. This PR makes the border radius for dark theme equal to that of light theme.

[CZO](https://chat.zulip.org/#narrow/channel/9-issues/topic/keys.20not.20centered.20in.20help.20menu/near/2077139)

<table>
<tr>
<th>
Before
</th>
<th>
After
</th>
</tr>
<th colspan="2">
Help menu/shortcut keys/The basics
</th>
<tr>
<td>

![image](https://github.com/user-attachments/assets/b9026777-1716-48e6-abd2-002090bba1a5)

</td>
<td>

![image](https://github.com/user-attachments/assets/c857313e-10a7-4962-b8df-8184050664f6)

</td>
</tr>
<th colspan="2">
Help menu/shortcut keys/Search
</th>
<tr>
<td>

![image](https://github.com/user-attachments/assets/4b5eb2e5-3dd0-464d-af26-5bb471be409e)

</td>
<td>

![image](https://github.com/user-attachments/assets/1c4bfa3a-e07a-428d-88e4-c3874396aec2)

</td>

</tr>
<th colspan="2">
Help menu/shortcut keys/Navigation
</th>
<tr>
<td>

![image](https://github.com/user-attachments/assets/6b6df60d-710b-4a4d-9d2a-115f91174ce6)


</td>
<td>

![image](https://github.com/user-attachments/assets/1743d638-b118-42d1-913f-8c38964c0bca)

</td>

</tr>
<th colspan="2">
Help menu/shortcut keys/Composing Messages
</th>
<tr>
<td>

![image](https://github.com/user-attachments/assets/b32bb6c7-83c9-4fa3-ad39-8fc0da48407d)

</td>
<td>

![image](https://github.com/user-attachments/assets/ffb0e681-d0b1-403b-bb7f-26cddf3e8ae1)

</td>
</tr>
<th colspan="2">
Help menu/shortcut keys/Recent Converations
</th>
<tr>
<td>

![image](https://github.com/user-attachments/assets/bf5e44a2-6d1e-4299-83b3-143b271a6394)

</td>
<td>

![image](https://github.com/user-attachments/assets/6d4552b7-482d-4ab9-bccc-932bb6473cd4)

</td>
</tr>
<th colspan="2">
Help menu/shortcut keys/Channel Settings
</th>
<tr>
<td>

![image](https://github.com/user-attachments/assets/e2e42960-a6e5-49a8-a32c-d87ff60d1deb)


</td>
<td>

![image](https://github.com/user-attachments/assets/65a06cb9-b61a-43b0-90a3-14b981eafda0)

</td>

</tr>
<th colspan="2">
Help menu/Message Formatting
</th>
<tr>
<td>

![image](https://github.com/user-attachments/assets/7a3a3fd8-8649-46bf-b4a3-346274b405c9)

</td>
<td>

![image](https://github.com/user-attachments/assets/ee0eb96f-8f99-403f-b5f9-c868eb6f5b2a)

</td>
</tr>
</table>

<details>
<summary>Self-review checklist</summary>

<!-- Prior to submitting a PR, follow our step-by-step guide to review your own code:
https://zulip.readthedocs.io/en/latest/contributing/code-reviewing.html#how-to-review-code -->

<!-- Once you create the PR, check off all the steps below that you have completed.
If any of these steps are not relevant or you have not completed, leave them unchecked.-->

- [x] [Self-reviewed](https://zulip.readthedocs.io/en/latest/contributing/code-reviewing.html#how-to-review-code) the changes for clarity and maintainability
      (variable names, code reuse, readability, etc.).

Communicate decisions, questions, and potential concerns.

- [x] Explains differences from previous plans (e.g., issue description).
- [x] Highlights technical choices and bugs encountered.
- [x] Calls out remaining decisions and concerns.
- [x] Automated tests verify logic where appropriate.

Individual commits are ready for review (see [commit discipline](https://zulip.readthedocs.io/en/latest/contributing/commit-discipline.html)).

- [x] Each commit is a coherent idea.
- [x] Commit message(s) explain reasoning and motivation for changes.

Completed manual review and testing of the following:

- [x] Visual appearance of the changes.
- [x] Responsiveness and internationalization.
- [x] Strings and tooltips.
- [x] End-to-end functionality of buttons, interactions and flows.
- [x] Corner cases, error conditions, and easily imagined bugs.
</details>

alya commented:
That seems better, though I think the vertical alignment is still off (too high). Can you take a look at that while we're here?

alya commented:
You also need to update your commit message to match the changes (3px, not 4).

whilstsomebody commented:
Hey @alya , I have addressed your comments. Could you please take a look?

whilstsomebody commented:
Hey @alya , could you please take a look?

alya commented:
Can you please take screenshots that are *identical* other than the changes you made, so that they are easy to compare by flipping back and forth?

whilstsomebody commented:
> Can you please take screenshots that are _identical_ other than the changes you made, so that they are easy to compare by flipping back and forth?

@alya , could you please take a look? I have added the screenshots in the description.

alya commented:
The first before/after pair looks like a nice improvement... but I don't understand what the rows in your table are. Can you label them?

whilstsomebody commented:
Hey @alya , could you see if now the labels are fine?

whilstsomebody commented:
Hey @alya , can you see the description if it's fine now?

alya commented:
Oh, I see, so this is just scrolling the menu, all at the same font size.

alya commented:
The screenshots look good to me, but I haven't tested.

@karlstolley could you please take a look?

karlstolley commented on web/styles/dark_theme.css:
```diff
@@ -47,9 +47,7 @@
     #keyboard-shortcuts {
         & kbd {
             border: 1px solid var(--color-hotkey-hint);
```
Seems like while you're working here, we could clean up and remove this whole block. Presumably there are already `border` and `color` values set for light mode on the `--color-hotkey-hint` variable. If the opacity is different, then establishing a quick `--opacity-hotkey-hint` should do the trick.

whilstsomebody replied:
Okh so I have done this.

Used `--color-hotkey-hint` for color and border.  [Here]()
Introduced a new variable `--opacity-hotkey-hint` ro handle the opacity for both the themes. [here]()

Updated the screenshots in the PR description.

karlstolley commented on web/templates/keyboard_shortcuts.hbs:
```diff
@@ -10,7 +10,7 @@
                 <tr>
                     <td class="definition">{{t 'Reply to message' }}</td>
-                    <td><span class="hotkey"><kbd>Enter</kbd> or <kbd>R</kbd></span></td>
+                    <td><span class="hotkey"><kbd><span>Enter</span></kbd> or <kbd>R</kbd></span></td>
```
Why is it necessary to introduce an anonymous `<span>` here?

whilstsomebody replied:
Oops, I think that was overseen by me while I was testing different approaches. I removed that.

whilstsomebody commented:
Hey @karlstolley , I have addressed your comments.

All the three kind of changes, i.e., Equalizing the border-radius for the theme, removing the redundant code and center aligning the text, have been handled in three different commits in the same order.

Could you please take a look?

whilstsomebody commented:
Hey @karlstolley , I have addressed your comments.

All the three kind of changes, i.e., Equalizing the border-radius for the theme, removing the redundant code and center aligning the text, have been handled in three different commits in the same order.

Could you please take a look?

karlstolley commented:
@alya how are you feeling about the change in border-color here?

alya commented:
The color change is a completely separate question we have no good reason to engage with at the moment. I don't even want to spend time thinking about it this close to the release.

whilstsomebody commented:
> The color change is a completely separate question we have no good reason to engage with at the moment. I don't even want to spend time thinking about it this close to the release.

I have reverted the color change of the kbd buttons.

karlstolley commented on web/styles/dark_theme.css:
```diff
@@ -47,10 +47,7 @@
     #keyboard-shortcuts {
         & kbd {
             border: 1px solid var(--color-hotkey-hint);
-            border-radius: 3px;
```
This border-radius should've been removed in the commit where it's specified in light theme.

karlstolley commented on web/styles/informational_overlays.css:
```diff
@@ -73,8 +73,8 @@
-        line-height: 1;
-        padding: 0 0.2em 0.2em;
+        line-height: normal;
+        padding: 0 0.2em;
```
I don't understand what's being achieved here or in `app_components` with zeroing out the bottom padding. I'm seeing numerous keys that appear to be too low in their box, and removing the bottom padding seems like a likely culprit. And `normal` as a line-height value is much more difficult to reason about when trying to do precision alignment.

karlstolley replied:
![Screenshot 2025-02-19 at 4 29 39 PM](https://github.com/user-attachments/assets/6456ba5e-c5b6-49ad-b48a-9e27acd87627)

whilstsomebody commented:
@karlstolley , I found that this issue is not observed in chrome .

![image](https://github.com/user-attachments/assets/9b20d6c4-b4c3-421a-b04e-168480c2be93)

But, in firefox, it does.

![image](https://github.com/user-attachments/assets/c46572c3-9d27-4f22-aef5-46ac80997a9f)

alya commented:
OK, it needs to look good across different modern browsers, not just in Chrome.
